code/game/machinery/baseball_machine.dm

```diff
@@ -0,0 +1,82 @@
+/obj/machinery/baseball_machine
+	name = "baseball pitching machine"
+	desc = "Hey batter batter!"
+	icon = 'icons/obj/turrets.dmi'
+	icon_state = "turretCover"
+	anchored = TRUE
+	density = TRUE
+	use_power = IDLE_POWER_USE				//this turret uses and requires power
+	idle_power_usage = 50		//when inactive, this turret takes up constant 50 Equipment power
+	active_power_usage = 300	//when active, this turret takes up constant 300 Equipment power
+	power_channel = EQUIP	//drains power from the EQUIPMENT channel
+	var/baseball_count = 15 // starts with 15 baseballs so we don't just spawn INFINITE amounts of baseballs
+	var/active = FALSE
+
+/obj/machinery/baseball_machine/Initialize(mapload)
+	. = ..()
+	for(var/x in 1 to baseball_count)
+		new /obj/item/beach_ball/holoball/baseball(src)
+
+/obj/machinery/baseball_machine/attackby(obj/item/I, mob/user, params)
+	if(default_unfasten_wrench(user, I))
+		return
+
+	if(istype(I, /obj/item/beach_ball))
+		if(!user.drop_item())
+			return
+		visible_message("<span class='notice'>[src] loads the baseball machine with a [I].</span>")
```

Comment:
So you can load an unlimited amount of balls into this thing?

Reply:
If you can get unlimited balls yes, but you’re not given more than 15 to start